test/ItspServices.pServer.ServerTest/Persistence.SqliteTests/UserRepositoryTests.cs

```diff
@@ -97,6 +101,66 @@ public void GetUserById_ShouldSucceed()
             expectedKeyData[2] = (byte) 't';
             expectedKeyData[3] = (byte) 'a';
             CollectionAssert.AreEquivalent(expectedKeyData, fooUser.PublicKeys[0].KeyData);
+
+            User barUser = repository.GetById(2);
+
+            Assert.AreEqual(2, barUser.Id);
+            Assert.AreEqual("BarUser", barUser.UserName);
+            Assert.AreEqual("BarUser".Normalize(), barUser.NormalizedUserName);
+            Assert.AreEqual("OtherPassword", barUser.PasswordHash);
+            Assert.AreEqual("Admin", barUser.Role);
+            Assert.AreEqual(0, barUser.PublicKeys.Count);
+        }
+
+        [TestMethod]
+        public void GetUserByInvalidId_ShouldReturnNull()
+        {
+            User invalidUser = repository.GetById(999);
+            Assert.IsNull(invalidUser);
+        }
+
+        [TestMethod]
+        public void AddUser_ShouldSucceed()
+        {
+            using (DbCommand insertTestData = memoryDbConnection.CreateCommand())
+            {
+                insertTestData.CommandText = "INSERT INTO Roles ('Name') VALUES ('User');";
+                insertTestData.ExecuteNonQuery();
+            }
+
+            using (IAddUnitOfWork<User> uow = repository.Add())
+            {
+                uow.Entity.UserName = "FooUser";
+                uow.Entity.NormalizedUserName = uow.Entity.UserName.Normalize();
+                uow.Entity.PasswordHash = "pw";
+                uow.Entity.Role = "User";
+                uow.Entity.PublicKeys = new List<Key>();
+                uow.Entity.PublicKeys.Add(new Key() { 
+                    KeyData = Encoding.UTF8.GetBytes("keydata"),
+                    Flag = Key.KeyFlag.ACTIVE
+                });
+
+                uow.Complete();
+            }
+
+            using (DbCommand queryData = memoryDbConnection.CreateCommand())
+            {
+                queryData.CommandText = "SELECT * FROM Users;";
+                using (IDataReader reader = queryData.ExecuteReader())
+                {
+                    reader.Read();
+                    Assert.AreEqual(1, reader.GetInt32(0));
+                    Assert.AreEqual("FooUser", reader.GetString(1));
+                    Assert.AreEqual("pw", reader.GetString(2));
+                    Assert.AreEqual(1, reader.GetInt32(3));
```

Comment:
Maybe you can add here a `Assert.IsFalse(reader.Read())` because you should only get one User, more then one would be an error, right?

Reply:
Resolved on line 156.